### PR TITLE
[parametric] Parametric go example

### PR DESCRIPTION
**Pull request by d4hines**
## Depends

- [ ] #649 

## Problem

We need some example Go code. 

## Solution

Adds a counter example that mirrors the JS example.

### CODE REVIEW

File: examples/go-counter/main.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/marigold-dev/deku/sdks/deku_go_interop"
+)
+
+type message struct {
```
Comment (gsebil08) [inline examples/go-counter/main.go:12]: What's the point of having a `struct` with only one field? Isn't it ok to use
```go
type message string
```
?

File: examples/go-counter/main.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+				tx_hash,
+				string(input)))
+		err := json.Unmarshal(input, &message)
+		check(err)
+		counter_bytes := deku_go_interop.Get("counter")
+		var counter *int
+		json.Unmarshal(counter_bytes, &counter)
+		var new_counter int
+		switch message.Action {
+		case "Increment":
```
Comment (gsebil08) [inline examples/go-counter/main.go:42]: Why pattern-matching on a string? Isn't it possible to use a sum-type? _(I don't know go, this is a naive question)_

**Review by gsebil08 — COMMENTED** [review]
Why did I read go?...

File: examples/go-counter/main.go (hunk 2 above)
Comment (d4hines) [reply examples/go-counter/main.go:42]: Go does not have easy sum types.

File: examples/go-counter/main.go (hunk 1 above)
Comment (Pilou97) [reply examples/go-counter/main.go:12]: Because the message is a json like message, and it has the following shape:
```json
{
"Action": "Increment"
}
```

Comment (oteku) [inline examples/go-counter/main.go:12]: Whaever is our opinon on Go  🤫
I think we should use the go-ish way to manage enumeration in our exemple... only for marketing purpose

```go
type Action int

const (
	Decremenet Action = iota+1
	Increment 
)

type Message struct {
	Action Action
}
```

**Pull request closed without merging**